kernel/debugmalloc.c

```diff
@@ -393,10 +414,39 @@ int myst_debug_posix_memalign(void** memptr, size_t alignment, size_t size)
     const size_t block_size = _calculate_block_size(alignment, size);
     void* block = NULL;
     header_t* header = NULL;
+    size_t rsize = _round_up_to_multiple(size, sizeof(uint64_t));
+
+    if (memptr)
+        *memptr = NULL;
 
     if (!memptr)
         return EINVAL;
 
+    /*
+    ** [padding][header][block][footer]
+    ** ^                ^
+    ** |                |
+    ** X                Y
+    **
+    ** Note: both X and Y are on the alignment boundary
```

Comment:
Who do we need to align the header, but not the footer?

Reply:
Good catch! We align the footer on an 8-byte-boundary but my picture did not reflect that. I updated the picture and added some more explanation.